checker/src/org/checkerframework/checker/formatter/FormatterTreeUtil.java

```diff
@@ -386,12 +386,17 @@ public AnnotationMirror categoriesToFormatAnnotation(ConversionCategory[] args)
 
     /**
      * Takes a syntax tree element that represents a {@link Format} annotation,
-     * and returns its value.
+     * and returns its value. Returns null (instead of an empty array) if the
+     * {@code Format} annotation's value is empty.
      */
     public ConversionCategory[] formatAnnotationToCategories(AnnotationMirror anno) {
+        AnnotationValue av = AnnotationUtils.getElementValuesWithDefaults(anno).get(formatArgTypesElement);
+        if (av == null) {
+        	return null;
```

Comment:
This is just pushing the NPE elsewhere.  The format annotation should always have a value, so if it does not, there's a bug somewhere.  Instead of returning null, Error.abort with a descriptive message.